src/marshmallow_sqlalchemy/schema.py

```diff
@@ -181,6 +181,16 @@ def get_instance(self, data):
             return self.session.query(self.opts.model).filter_by(**filters).first()
         return None
 
+    @ma.pre_load
+    def set_field_session(self, data, **kwargs):
+        """Pass session to nested fields."""
+        nested_fields = {
```

Comment:
This intermediate dictionary is unnecessary. You can just iterate through `self.fields.values()` with an `if`.